SCons/Variables/__init__.py

```diff
@@ -287,7 +290,13 @@ def Update(self, env, args: Optional[dict] = None) -> None:
         for option in self.options:
             if option.validator and option.key in values:
                 if option.do_subst:
-                    value = env.subst('${%s}' % option.key)
+                    val = env[option.key]
+                    if not SCons.Util.is_String(val):
```

Comment:
This could be more specific and check for an instance of `_ListVariable`, but that's supposed to be private to the ListVariable module and the main Variables package isn't supposed to have to know details...